src/panels/lovelace/editor/config-elements/hui-glance-card-editor.ts

```diff
@@ -20,13 +21,35 @@ import "../../components/hui-entity-editor";
 import "../../../../components/ha-card";
 import "../../../../components/ha-icon";
 
+const entitiesConfigStruct = struct.union([
+  {
+    entity: "entity-id",
+    name: "string?",
+    icon: "icon?",
+  },
+  "entity-id",
+]);
+
+const cardConfigStruct = struct({
+  type: "string",
+  id: "string|number",
+  title: "string|number?",
+  theme: "string?",
+  columns: "number?",
+  show_name: "boolean?",
+  show_state: "boolean?",
+  entities: [entitiesConfigStruct],
+});
+
 export class HuiGlanceCardEditor extends hassLocalizeLitMixin(LitElement)
   implements LovelaceCardEditor {
   public hass?: HomeAssistant;
   private _config?: Config;
   private _configEntities?: ConfigEntity[];
 
   public setConfig(config: Config): void {
+    config = cardConfigStruct(config);
```

Comment:
Should we be manipulating the parameter value or just use the right hand in line 53?

Reply:
It just checks if it is valid, if so returns the value (you can also set default values, but in this case it will not alter anything) or it will throw.